src/fsharp/TypeChecker.fs

```diff
@@ -4933,18 +4949,101 @@ and TcStaticConstantParameter cenv (env: TcEnv) tpenv kind (v: SynType) idOpt co
                 | _ -> fail()
             | _ -> error(Error(FSComp.SR.tcInvalidConstantExpression(), v.Range))
         v, tpenv'
-    | SynType.LongIdent lidwd ->
+    | SynType.LongIdent(lidwd) ->
         let m = lidwd.Range
-        TcStaticConstantParameter cenv env tpenv kind (SynType.StaticConstantExpr(SynExpr.LongIdent (false, lidwd, None, m), m)) idOpt container
-    | _ ->  
+        // let tyid = mkSynId tyid.idRange n
+        let (LongIdentWithDots(lid, _)) = lidwd
+        let ad = env.eAccessRights
+        match ResolveTypeLongIdent cenv.tcSink cenv.nameResolver ItemOccurence.UseInAttribute OpenQualified env.eNameResEnv ad lid TypeNameResolutionStaticArgsInfo.DefiniteEmpty  PermitDirectReferenceToGeneratedType.No with
```

Comment:
So I think here we are starting to resolve a specified type for a type provider.